### PR TITLE
Require Dailies to have a Start Date

### LINKED ISSUE #7715: Start Date can be set to null / empty string on a Daily
When editing a Daily on the website, you can delete the Start Date. After saving, the task has a Start Date of `null`.

This is confusing because it's not obvious to the player how cron will treat it (immediately due versus never due) and because the official and unofficial site/apps/tools will display the due-ness of the Daily in different ways (e.g., website says not due, Android app says due).

I believe the API should prohibit a non-existent or nonsensical Start Date. It should always be a valid date, and if a valid date isn't supplied, then default to the current date, which is what cron does anyway:
common/script/cron.js:  `let taskStartDate = moment(dailyTask.startDate).zone(o.timezoneOffset);`

The website should also handle deletion of the Start Date in a more informative way. For example, if there's no Start Date then don't let the Daily be saved, or save it but set the Start Date to today so that you see the date if you edit the task again, or set the Start Date to today as soon as you move focus out of the Start Date field. I think the latter is my favourite option at the moment.

### PR DESCRIPTION
**Pull request by librarianmage**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #7715 

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

[//]: # (Put User ID in here - found in Settings -> API)

----
UUID:

### CODE REVIEW

File: website/views/shared/tasks/edit/repeatables.jade [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
-      label
-        input(type="radio", ng-model='task._edit.repeatsOn', value='dayOfMonth')
-        =env.t('dayOfMonth')
-      label
-        input(type="radio", ng-model='task._edit.repeatsOn', value='dayOfWeek')
-        =env.t('dayOfWeek')
-
-    //- .form-group
-    //-   legend.option-title=env.t('summary')
+fieldset.option-group.advanced-option(ng-show="task.type === 'daily'")
```
Comment (paglias) [inline website/views/shared/tasks/edit/repeatables.jade]: what changed in this file? Can't see from git diff

File: website/server/models/task.js [hunk 2]
```diff
@@ -229,6 +229,7 @@ export let DailySchema = new Schema(_.defaults({
     default () {
       return moment().startOf('day').toDate();
     },
+    required: true,
```
Comment (paglias) [inline website/server/models/task.js:232]: Test needed

**Review by paglias — CHANGES_REQUESTED** [review]

File: website/views/shared/tasks/edit/repeatables.jade (hunk 1 above)
Comment (MathWhiz) [inline website/views/shared/tasks/edit/repeatables.jade]: I

Comment (librarianmage) [reply website/views/shared/tasks/edit/repeatables.jade]: I can't tell either

Comment (khipkin) [reply website/views/shared/tasks/edit/repeatables.jade]: Append `?w=1` to the URL to ignore whitespace changes, and you'll see that only whitespace changed.

**Comment by librarianmage** [comment]
@paglias Ping!

**Comment by paglias** [comment]
This looks good but it'll probably require a migration for tasks that don't have the field unless we want to give it a default value. Ideas?

**Review by paglias — APPROVED** [review]

File: website/views/shared/tasks/edit/repeatables.jade (hunk 1 above)
Comment (Alys) [reply website/views/shared/tasks/edit/repeatables.jade]: The file has Windows line breaks in it in `develop`, so I'm guessing this change removes them and uses just Unix line breaks, which is consistent with the majority of other files in the repo.

**Comment by Alys** [comment]
There's some discussion of the default value in the issue although I'm thinking now that for simplicity and speed we should just go with a default value of "today".

**Comment by librarianmage** [comment]
@paglias @Alys What should I change?

**Comment by Alys** [comment]
@MathWhiz sorry about the delay on your question. My feeling is that with the code that you have now, you probably don't need to change anything regarding the default date. When a new Daily is created, it is always given the correct default date (today) so that part is fine as it is. The `required: true` that you've added should prevent the date being removed. I believe the `type: Date` code that was already there should prevent someone saving a value that isn't a date. So with this PR I think the API has everything it needs.

Ideally, there'd be code on the website to give a friendly error message to the user if they try to delete the start date, but since we're rewriting the website and since deleting the start date is probably rare, I don't think it's a critical thing to add now.

**Comment by librarianmage** [comment]
Does that meat that everything looks good?

**Review by paglias — APPROVED** [review]
Can be merged for me!

**Comment by librarianmage** [comment]
Ping? I am really sorry for bugging you all, but I would really like to know if I am missing anything.

**Comment by paglias** [comment]
Notated towards your next tier! Thanks and sorry again for the super long delay!

**Comment by librarianmage** [comment]
Cool, thanks!

**Pull request merged**